src/gobexport/csv_inspector.py

```diff
@@ -126,10 +150,12 @@ def _check_columns(self, columns: list[str], line_no: int):
 
     def check_lines(self, lines):
         # Start at line 1 (skip header) and stop at end of lines. Skip any (possibly trailing) empty line
-        for (line_idx, line) in [(i, l) for (i, l) in enumerate(lines[1:]) if l]:
-            columns = line.strip().split(";")
-            self._check_columns(columns, line_idx + 2)  # +2 = 1 for 0 offset and 1 for skipped header
+        for line_idx, line in enumerate(lines):
+            if line and line_idx > 0:
+                columns = line.split(";")
+                self._check_columns(columns, line_idx + 1)  # +1 = 1 for 0 offset
 
         self._check_uniqueness()
+        self.unique_values.clear()  # clear refs
```

Comment:
Why? As soon as the object is out of scope the dict should be destroyed by Python, right?

Now you're assuming check_lines is the last method called, because the method will misbehave when check_lines is called again. This class should not worry about how it's being used though.